apps/transport/lib/http/utils.ex

```diff
@@ -2,10 +2,71 @@ defmodule Transport.Http.Utils do
   @moduledoc """
   Useful functions to work with http requests
   """
+  require Logger
 
-  def location_header(headers) do
-    for {key, value} <- headers, String.downcase(key) == "location" do
+  def location_header(headers), do: header_value(headers, "location")
+
+  def header_value(headers, header) do
+    for {key, value} <- headers, String.downcase(key) == String.downcase(header) do
       value
     end
   end
+
+  @doc """
+  Takes a response body and response headers and try to convert the body to UTF-8.
+  It looks at the charset defined in the `content-type` header.
+
+  Inspired by https://bernheisel.com/blog/httpoison-and-decompression
+  """
+  def reencode_body_to_utf8(body, headers) do
+    headers
+    |> header_value("content-type")
+    |> parse_charset()
+    |> reencode_body(body)
+  end
+
+  defp parse_charset([content_type]), do: parse_charset(content_type)
+
+  defp parse_charset(content_type) when is_binary(content_type) do
+    case Plug.Conn.Utils.content_type(content_type) do
+      {:ok, _, _, %{"charset" => charset}} ->
+        # Case insensitive and ignore dashes
+        cond do
+          charset =~ ~r/utf-?8/i -> :utf8
+          charset =~ ~r/iso-?8859-?1/i -> :latin1
+          true -> charset
+        end
+
+      _ ->
+        nil
+    end
+  end
+
+  defp parse_charset(_), do: nil
+
+  # When the header isn't sent, the RFC spec says we should assume ISO-8859-1, but the default is
+  # actually different per format, eg, XML should be assumed UTF-8. We're going to not re-encode
+  # if it's not sent and assume UTF-8. This should be safe for most cases.
+  defp reencode_body(nil, body), do: body
+  defp reencode_body(:utf8, body), do: body
+
+  defp reencode_body(:latin1, body) do
```

Comment:
```suggestion
  # Re-encode the body to ISO-8859-1/Latin1. Current limitation (because we haven't seen the need yet):
  # we do not make any attempt to modify the XML prolog (e.g. `<?xml version="1.0" encoding="UTF-8"?>`)
  # to match the target encoding. If the prolog is sent on some pails, we will have to "string replace" that part
  # too.
  defp reencode_body(:latin1, body) do
```

Reply:
Je prends pas la suggestion telle quelle :
- c'est _from_ Latin 1 to UTF-8
- la remarque sur le XML prolog a plus sa place dans le `SIRIQuerier` que dans le module courant qui est générique, je déplace là-bas

Reply:
Oui pardon ! Faute de frappe

Reply:
> la remarque sur le XML prolog a plus sa place dans le SIRIQuerier que dans le module courant qui est générique, je déplace là-bas

Je parlais bien de l'analyse de la réponse, si le prologue est éventuellement fourni ! Mais on verra si ça arrive, pas de souci.

Reply:
Ok vu plus bas, merci !